cmd/main_test.go

```diff
@@ -267,3 +270,83 @@ var _ = AfterSuite(func() {
 		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error killing the server during test teardown: %v", err))
 	}
 })
+
+type mockAuthorization struct {
+	allowed bool
+	err error
+}
+
+func (m *mockAuthorization) runAccessCheck(
+	e *echo.Echo,
+	authCl authorizationv1Client.AuthorizationV1Interface,
+	email, namespace, resourceGroup, resource, verb string,
+) (bool, error) {
+	return m.allowed, m.err
+}
+
+var _ = Describe("TestGetNamespacesWithAccess", func() {
+	var (
+		e               *echo.Echo
+		c               echo.Context
+		authCl          *mockAuthorization
+		allNamespaces   []k8sapi.Namespace
+		expectedNs      []k8sapi.Namespace
+		actualNs        []k8sapi.Namespace
+		err             error
+	)
+
+	BeforeEach(func() {
+		e = echo.New()
+		req := httptest.NewRequest(http.MethodGet, "/", nil)
+		rec := httptest.NewRecorder()
+		c = e.NewContext(req, rec)
+		c.Request().Header.Set("test-email", "test@example.com")
+
+		authCl = &mockAuthorization{}
```

Comment:
I'm not aware of a way to mock like this, is it working?
anyway, I'm not sure we can mock it - that's why we are using the testenv. 
can just create one using `	clientset.AuthorizationV1()` (take a look at `getWorkspacesWithAccess`)